bin/crew

```diff
@@ -718,13 +667,12 @@ def upgrade(*pkgs, build_from_source: false)
         return false
       end
     end
-    pkgVer_latest     = Package.load_package(pkgFile, pkgName).version
+    set_package File.join(CREW_PACKAGES_PATH, "#{pkgName}.rb")
     pkgVer_installed  = @device[:installed_packages].select { |pkg| pkg[:name] == pkgName } [0][:version]
-    pkgHash_latest    = Package.load_package(pkgFile, pkgName).get_binary_sha256(@device[:architecture])
     pkgHash_installed = @device[:installed_packages].select { |pkg| pkg[:name] == pkgName } [0][:binary_sha256]
 
-    return pkgHash_latest != pkgHash_installed unless !pkgHash_installed || pkgHash_latest == ''
-    return pkgVer_latest != pkgVer_installed
+    return @pkg.binary_sha256&.dig(ARCH.to_sym) != pkgHash_installed unless !pkgHash_installed || @pkg.binary_sha256&.dig(ARCH.to_sym) == 'null'
```

Comment:
Is there any case that `binary_sha256` is `'null'`?

Reply:
Yes-- if binary_sha256 is not found when writing to json, the ruby nil will become a string literall null in the json file